src/MSBuild/XMake.cs

```diff
@@ -866,8 +868,7 @@ private static void Console_CancelKeyPress(object sender, ConsoleCancelEventArgs
         {
             if (e.SpecialKey == ConsoleSpecialKey.ControlBreak)
             {
-                e.Cancel = false; // required; the process will now be terminated rudely
-                return;
+                Environment.Exit(1); // the process will now be terminated rudely
```

Comment:
It looks like this code runs in the client, right? Does this make it exit fast enough that the server doesn't know? I'm also wondering if there's something similar with normal (serverless) builds.

Reply:
I tried Ctrl-Break in "normal" build and for some reason, it didn't work (the process wasn't terminated). I guess this functionality is not used heavily, so there was no reported issue.
When client process terminates during server mode build, the pipe connection is broken and it causes unhandled exception in server and server is terminated too.